### PR TITLE
[Tests] Add unit test for InternalAdjecencyMatrix aggregation

### PR DESCRIPTION
**Pull request by cbuescher**
Adding a unit test to InternalAdjecencyMatrix that extends the shared InternalAggregationTestCase that we use for testing aggregations. 
Relates to #22278

### CODE REVIEW

**Review by javanna — APPROVED** [review]
LGTM

**Comment by cbuescher** [comment]
@javanna thanks for the quick review, will wait for CI before I merge

**Pull request merged**
